### PR TITLE
Migrated KSQL article

### LINKED ISSUE #906: Migrate KSQL on Aiven for Apache Kafka
# What should we add to our content collections?

(more information is useful here! Were you trying to do something particular?)

Migrate 

https://help.aiven.io/en/articles/2281428-using-ksql-with-aiven-for-apache-kafka

# Is this something you would work on yourself?

[X] I plan to open a pull request for this

### PR DESCRIPTION
**Pull request by ftisiot**
# What changed, and why it matters

Migrated https://help.aiven.io/en/articles/2281428-using-ksql-with-aiven-for-apache-kafka

Fixes #906

### CODE REVIEW

**Comment by encima** [comment]
Is it worth calling out Karapace here and recommended usage of that over Confluent (due to platform compatibility and licensing)?

**Comment by laysauchoa** [comment]
Who can help you with the review?

**Comment by ftisiot** [comment]
Added mentions to Karapace.
Will ask the help of SAs to test!

File: docs/products/kafka/howto/ksql-docker.rst [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+        > ca_schema_registry.cert
+
+2. Create the truststore with the following ``keytool`` command  by replacing the ``TRUSTSTORE_SCHEMA_REGISTRY_FILE_NAME`` placeholder::
+
+    keytool -import -file ca_schema_registry.cert \
+        -alias CA \
+        -keystore \
+        TRUSTSTORE_SCHEMA_REGISTRY_FILE_NAME
+   
+   You'll be prompted to insert a password, which in this example is referenced by the placeholder ``TRUSTSTORE_SCHEMA_REGISTRY_PASSWORD``
```
Comment (runwuf) [inline docs/products/kafka/howto/ksql-docker.rst:52]: Password prompt can be skipped by adding `-noprompt` and `-storepass` options provided.

    keytool -import -file ca_schema_registry.cert \
        -alias CA \
        -keystore TRUSTSTORE_SCHEMA_REGISTRY_FILE_NAME \
        -storepass TRUSTSTORE_SCHEMA_REGISTRY_PASSWORD \
        -noprompt

File: docs/products/kafka/howto/ksql-docker.rst [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+            -Dssl.truststore.type=JKS
+            -Dssl.truststore.location=/ssl_settings/TRUSTSTORE_FILE_NAME
+            -Dssl.truststore.password=SSL_TRUSTSTORE_PASSWORD
+            -Dksql.schema.registry.url=APACHE_KAFKA_HOST:SCHEMA_REGISTRY_PORT
+            -Dksql.schema.registry.basic.auth.credentials.source=USER_INFO
+            -Dksql.schema.registry.basic.auth.user.info=avnadmin:SCHEMA_REGISTRY_PASSWORD
+            -Dksql.schema.registry.ssl.truststore.location=/ssl_settings/TRUSTSTORE_SCHEMA_REGISTRY_FILE_NAME
+            -Dksql.schema.registry.ssl.truststore.password=TRUSTSTORE_SCHEMA_REGISTRY_PASSWORD" \
+        confluentinc/ksqldb-server:0.23.1
+
```
Comment (runwuf) [inline docs/products/kafka/howto/ksql-docker.rst:99]: Some docker setup has issues using mounting options `-v`
In my case using `minikube + hyperkit + dockercli` it's a pain... Perhaps provide alternative option here just copy the keystores into the container?

```
docker container create --name ksql  \
    -p 127.0.0.1:8088:8088 \
    -e KSQL_BOOTSTRAP_SERVERS=APACHE_KAFKA_HOST:APACHE_KAFKA_PORT \
    -e KSQL_LISTENERS=http://0.0.0.0:8088/ \
    -e KSQL_KSQL_SERVICE_ID=ksql_service_1_ \
    -e KSQL_OPTS="-Dsecurity.protocol=SSL
        -Dssl.keystore.type=PKCS12
        -Dssl.keystore.location=/home/appuser/KEYSTORE_FILE_NAME
        -Dssl.keystore.password=SSL_KEYSTORE_PASSWORD
        -Dssl.key.password=SSL_KEY_PASSWORD
        -Dssl.truststore.type=JKS
        -Dssl.truststore.location=/home/appuser/TRUSTSTORE_FILE_NAME
        -Dssl.truststore.password=SSL_TRUSTSTORE_PASSWORD
        -Dksql.schema.registry.url=APACHE_KAFKA_HOST:SCHEMA_REGISTRY_PORT
        -Dksql.schema.registry.basic.auth.credentials.source=USER_INFO
        -Dksql.schema.registry.basic.auth.user.info=avnadmin:SCHEMA_REGISTRY_PASSWORD
        -Dksql.schema.registry.ssl.truststore.location=/home/appuser/TRUSTSTORE_SCHEMA_REGISTRY_FILE_NAME
        -Dksql.schema.registry.ssl.truststore.password=TRUSTSTORE_SCHEMA_REGISTRY_PASSWORD" \
    confluentinc/ksqldb-server:0.23.1
docker cp KEYSTORE_FILE_NAME ksql:/home/appuser/
docker cp TRUSTSTORE_FILE_NAME ksql:/home/appuser/
docker cp TRUSTSTORE_SCHEMA_REGISTRY_FILE_NAME ksql:/home/appuser/
docker start ksql
```

**Comment by runwuf** [comment]
@ftisiot I've tested all the steps and provided two minor suggestions, I confirmed it works as expected.

**Review by runwuf — CHANGES_REQUESTED** [review]

File: docs/products/kafka/howto/ksql-docker.rst (hunk 1 above)
Comment (ftisiot) [reply docs/products/kafka/howto/ksql-docker.rst:52]: Right thanks!

**Review by runwuf — APPROVED** [review]

**Comment by ftisiot** [comment]
@TibsAtWork tagged you here for one last check regarding language before we're ready to merge. @runwuf already checked it on the technical side

**Comment by TibsAtWork** [comment]
A few minor english changes, and sorting out that pesky internal link to make vale happy.

**Review by TibsAtWork — APPROVED** [review]
I think it's OK

**Pull request merged**